sbnana/CAFAna/XSec/Flux.cxx

```diff
@@ -126,4 +185,67 @@ namespace ana
     ret->GetYaxis()->SetTitle(ytitle.c_str());
     return ret;
   }
+
+  //----------------------------------------------------------------------
+  void EnsembleFluxTimesNuclei::SaveTo(TDirectory* dir, const std::string& name) const
+  {
+    TDirectory* tmp = gDirectory;
+
+    dir = dir->mkdir(name.c_str()); // switch to subdir
+    dir->cd();
+
+    TObjString("EnsembleFluxTimesNuclei").Write("type");
+
+    TVectorD pdg(1);
+    pdg[0] = fPdg;
+    pdg.Write("pdg");
+
+    EnsembleSpectrum::SaveTo(dir, "ensemblespectrum");
+
+    dir->Write();
+    delete dir;
+
+    tmp->cd();
+  }
+
+  //----------------------------------------------------------------------
+  std::unique_ptr<EnsembleFluxTimesNuclei> EnsembleFluxTimesNuclei::LoadFrom(TDirectory* dir, const std::string& name)
+  {
+    dir = dir->GetDirectory(name.c_str()); // switch to subdir
+    assert(dir);
+
+    DontAddDirectory guard;
+
+    std::unique_ptr<TObjString> tag((TObjString*)dir->Get("type"));
+    assert(tag);
+    assert(tag->GetString() == "EnsembleFluxTimesNuclei");
+
+    std::unique_ptr<TVectorD> pdg((TVectorD*)dir->Get("pdg"));
+    assert(pdg && pdg->GetNrows()==1);
+
+    return std::unique_ptr<EnsembleFluxTimesNuclei>(new EnsembleFluxTimesNuclei(EnsembleSpectrum::LoadFrom(dir, "ensemblespectrum"), (*pdg)[0]));
+  }
+
+  //----------------------------------------------------------------------
+  EnsembleFluxTimesNuclei::EnsembleFluxTimesNuclei(const std::unique_ptr<EnsembleSpectrum> spec, const int pdg)
+    : EnsembleSpectrum(*spec), fPdg(pdg)
+  {
+  }
+
+  //----------------------------------------------------------------------
+  EnsembleSpectrum EnsembleFluxTimesNuclei::MakeTotalFlux(const LabelsAndBins& ax) const
+  {
+    const unsigned int nuniv = NUniverses();
+    const unsigned int nbins = ax.GetBins1D().NBins()+2;
+
+    Hist h = Hist::Zero(nbins * nuniv);
+
+    for(unsigned int univIdx = 0; univIdx < nuniv; ++univIdx){
+      const double univIntegral(Universe(univIdx).Integral(fPOT));
+      for(unsigned int bin = 0; bin < nbins; ++bin)
+        h.Fill(nbins * univIdx + bin, univIntegral);
+    }
+
+    return EnsembleSpectrum(fMultiverse, std::move(h), fPOT, fLivetime, LabelsAndBins(ax));
```

Comment:
Note I had to call the `LabelsAndBins` constructor here still because the constructor expects an rvalue reference, which is not the case for spectrum.